### ISSUE TITLE
Potential Error in SpanPredictor Codes

### ISSUE THREAD

**Issue by NoviScl** [shmsw25/AmbigQA#13]
In codes/models/span_predictor.py, the classifier layer is defined as:
`self.qa_classifier = nn.Linear(config.hidden_size, 1)`

But shouldn't it be:
`self.qa_classifier = nn.Linear(config.hidden_size, config.num_labels)` (where config.num_labels=2)?

**Comment by shmsw25** [comment]
Hi @NoviScl, thanks for asking. This line is in line with our intent -- each passage outputs one scalar value as its logit, which is softmax'ed over all passages to get a likelihood of each block being selected. This part is implemented [here](https://github.com/shmsw25/AmbigQA/blob/master/codes/util.py#L30).

Your idea on having a two-way classifier can classify each passage between "being selected" and "not being selected", which should work reasonably well. But we found our implementation works better than two-way classifier in our previous work.

**Comment by NoviScl** [comment]
Ah got it. Thanks for the super quick response.
